Tasks/Maven/maven.ps1

```diff
@@ -28,43 +40,27 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
 import-module "Microsoft.TeamFoundation.DistributedTask.Task.TestResults"
 
-if($jdkVersion -and $jdkVersion -ne "default")
-{
-    $jdkPath = Get-JavaDevelopmentKitPath -Version $jdkVersion -Arch $jdkArchitecture
-    if (!$jdkPath) 
-    {
-        throw "Could not find JDK $jdkVersion $jdkArchitecture, please make sure the selected JDK is installed properly"
-    }
-
-    Write-Host "Setting JAVA_HOME to $jdkPath"
-    $env:JAVA_HOME = $jdkPath
-    Write-Verbose "JAVA_HOME set to $env:JAVA_HOME"
-}
+. ./mavenHelper.ps1
 
-Write-Verbose "Running Maven..."
-Invoke-Maven -MavenPomFile $mavenPOMFile -Options $options -Goals $goals
+# Use a specific JDK
+ConfigureJDK $jdkVersion $jdkArchitecture
 
-# Publish test results files
-$publishJUnitResultsFromAntBuild = Convert-String $publishJUnitResults Boolean
-if($publishJUnitResultsFromAntBuild)
-{
-   # check for JUnit test result files
-    $matchingTestResultsFiles = Find-Files -SearchPattern $testResultsFiles
-    if (!$matchingTestResultsFiles)
-    {
-        Write-Host "No JUnit test results files were found matching pattern '$testResultsFiles', so publishing JUnit test results is being skipped."
-    }
-    else
-    {
-        Write-Verbose "Calling Publish-TestResults"
-        Publish-TestResults -TestRunner "JUnit" -TestResultsFiles $matchingTestResultsFiles -Context $distributedTaskContext
-    }    
-}
-else
+# Invoke MVN
+Write-Host "Running Maven..."
+Invoke-Maven -MavenPomFile $mavenPOMFile -Options $options -Goals $goals -ErrorVariable mavenError
+
+if ($mavenError)
```

Comment:
This is incorrect, even if maven script fails we want it to try publishing test results. If script stops running, test results would never be published in case of test failures.